.circleci/config.yml

```diff
@@ -119,6 +119,23 @@ jobs:
       - run: go run build/ci.go test -coverage
       - run: bash <(curl -s https://codecov.io/bash)
 
+  race:
+    executor: golang
+    resource_class: medium+
+    steps:
+      - attach_workspace:
+          at: ~/repos
+      - restore_cache:
+          keys:
+            - go-mod-v1-{{ checksum "go.sum" }}
+      - run: go run build/ci.go test -race
+      - store_artifacts:
+          path: /tmp/test-results
+          destination: raw-test-output
+
+      - store_test_results:
+          path: /tmp/test-results
+
```

Comment:
I'd be in favour of just adding the -race flag to the existing unit tests job. Is there a reason for making this a separate job?

Reply:
Also if you are going to add a new job then it needs to be added into the workflows job list at the end of the file. Currently the race job is not being run.

Reply:
> I'd be in favour of just adding the -race flag to the existing unit tests job. Is there a reason for making this a separate job?

No, I was following the pattern of `coverage`, which is just adding `-coverage`.

Reply:
@piersy Adding `-race` will cause 167 unit tests report **data race**. 😅

Reply:
@piersy How about we do this `-race` separate CI task as a temporary approach to avoid adding those **fail** noise to `unit tests` CI?

Reply:
I think in this case having a separate CI job for the race tests makes sense, and we can make it non required.

Reply:
Then we can work on fixing them.